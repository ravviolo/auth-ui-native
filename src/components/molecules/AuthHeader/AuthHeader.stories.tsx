import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { StoryScreen } from 'storybook/StoryScreen';

import { AuthHeader } from './AuthHeader';

storiesOf('AuthHeader', module)
  .addDecorator((getStory) => <StoryScreen center>{getStory()}</StoryScreen>)
  .add('Default', () => (
    <AuthHeader
      testID={text('testID', 'auth-header-test-id')}
      title={text('title', 'This is main header.')}
    />
  ));
