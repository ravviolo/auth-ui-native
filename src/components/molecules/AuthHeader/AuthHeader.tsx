import { Heading } from 'components/atoms';
import { View } from 'react-native';

interface Props {
  title: string;
  testID: string;
}

export const AuthHeader = ({ title, testID }: Props) => {
  return (
    <View testID={testID}>
      <Heading testID="auth-heading-test-id" title={title} />
    </View>
  );
};
