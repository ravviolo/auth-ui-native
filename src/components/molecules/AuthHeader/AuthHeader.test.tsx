import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { AuthHeader } from './AuthHeader';

import type { Props } from './AuthHeader';

describe('AuthHeader', () => {
  const props: Props = {
    testID: 'auth-header-test-id',
    title: 'Header',
  };

  it('should display heading with title text received via props', () => {
    render(<AuthHeader {...props} />);

    const authHeader = screen.getByTestId('auth-header-test-id');

    expect(authHeader.children[0]).toHaveTextContent('Header');
  });

  it('should match snapshot', () => {
    const tree = testRenderer.create(<AuthHeader {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
