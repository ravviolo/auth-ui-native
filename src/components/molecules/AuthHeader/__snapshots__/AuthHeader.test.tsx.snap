// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeader should match snapshot 1`] = `
<View
  testID="auth-header-test-id"
>
  <Text
    style={
      Object {
        "color": "#3D3D3D",
        "fontSize": 18,
        "fontWeight": "bold",
        "textTransform": "uppercase",
      }
    }
    testID="auth-heading-test-id"
  >
    Header
  </Text>
</View>
`;
